codegenerator/cli/src/hbs_templating/codegen_templates.rs

```diff
@@ -349,55 +337,120 @@ impl EntityRecordTypeTemplate {
+                            "(Utils.magic: HyperSyncClient.Decoder.decodedEvent => eventArgs)"
+                                .to_string(),
+                        decode_hyper_fuel_data_code: Self::DECODE_HYPER_FUEL_DATA_CODE.to_string(),
+                    })
+                } else {
+                    let template_params = params
+                        .iter()
+                        .map(|input| {
+                            let res_type = abi_to_rescript_type(&input.into());
+                            let js_name = input.name.to_string();
+                            EventParamTypeTemplate {
+                                res_name: RescriptRecordField::to_valid_res_name(&js_name),
+                                js_name,
+                                default_value_rescript: res_type.get_default_value_rescript(),
+                                default_value_non_rescript: res_type
+                                    .get_default_value_non_rescript(),
+                                res_type: res_type.to_string(),
+                                is_eth_address: res_type == RescriptTypeIdent::Address,
+                            }
+                        })
+                        .collect::<Vec<_>>();
+
+                    let data_type_expr = RescriptTypeExpr::Record(
+                        params
+                            .iter()
+                            .map(|p| {
+                                RescriptRecordField::new(
+                                    p.name.to_string(),
+                                    abi_to_rescript_type(&p.into()),
+                                )
+                            })
+                            .collect(),
+                    );
+
+                    Ok(EventTemplate {
+                        name,
+                        params: template_params,
+                        data_type: data_type_expr.to_string(),
+                        data_schema_code: data_type_expr.to_rescript_schema(),
+                        topic0: config_event.topic0.to_string(),
+                        convert_hyper_sync_event_args_code:
+                            Self::generate_convert_hyper_sync_event_args_code(params),
+                        decode_hyper_fuel_data_code: Self::DECODE_HYPER_FUEL_DATA_CODE.to_string(),
+                    })
+                }
+            }
+            EventPayload::Data(type_indent) => Ok(EventTemplate {
+                name,
+                params: vec![],
+                data_type: type_indent.to_string(),
+                data_schema_code: type_indent.to_rescript_schema(),
+                topic0: config_event.topic0.to_string(),
+                convert_hyper_sync_event_args_code:
+                    "(Utils.magic: HyperSyncClient.Decoder.decodedEvent => eventArgs)".to_string(),
+                decode_hyper_fuel_data_code: Self::DECODE_HYPER_FUEL_DATA_CODE.to_string(),
+            }),
```

Comment:
Actually this case can be covered as well. All we want to do on the match of these types is return a rescript type expression.

Reply:
It'll change in the future with the `decode_hyper_fuel_data_code` implementation.